src/python/pants/rules/core/generate_pants_ini.py

```diff
@@ -16,36 +16,50 @@
 from pants.engine.goal import Goal, GoalSubsystem
 from pants.engine.rules import goal_rule
 from pants.engine.selectors import Get
-from pants.version import VERSION as pants_version
+from pants.version import VERSION
 
 
 class GeneratePantsIniOptions(GoalSubsystem):
   """Generate pants.ini with sensible defaults."""
   name = 'generate-pants-ini'
 
+  @classmethod
+  def register_options(cls, register) -> None:
+    super().register_options(register)
+    register('--v2-only', type=bool, fingerprint=True, default=False,
+             help='Support only v2 rules. If unspecified, this repo will also run v1 tasks.')
+
 
 class GeneratePantsIni(Goal):
   subsystem_cls = GeneratePantsIniOptions
 
 
 @goal_rule
-async def generate_pants_ini(console: Console, workspace: Workspace) -> GeneratePantsIni:
+async def generate_pants_ini(console: Console, workspace: Workspace,
+                             options: GeneratePantsIniOptions) -> GeneratePantsIni:
   pants_ini_content = dedent(f"""\
     [GLOBAL]
-    pants_version: {pants_version}
+    pants_version: {VERSION}
+    v1: {'False' if options.values.v2_only else 'True'}
+    v2: True
+    v2_ui: {'True' if options.values.v2_only else 'False'}
+    backend_packages: []
```

Comment:
NB: This is a change in semantics from normal Pants. Normally, we do register several backends by default. This is deregistering those options and will require users to explicitly opt into them.

Generally, I think users should have to opt into things, but we don't have a good story for discovering what backends are available at the moment (opened https://github.com/pantsbuild/pants/issues/8992 to track this). 

I recommend changing both of these options to be `+[]`, rather than `[]`.

Reply:
I changed this on purpose. We want to move towards opt-in, and since this is for new repos only there is no backwards compatibility issue.  Right now, for example, you're opted in to every codegen backend, which in v1 just generates noise. I can add some commented-out backends, that can serve as documentation for opting in.

Reply:
Good idea. I will also feel much better when we do #8992, which doesn’t seem very difficult.